cairo1-run/src/main.rs

```diff
@@ -475,6 +468,50 @@ mod tests {
         assert_matches!(run(args), Ok(Some(res)) if res == expected_output, "Program {} failed with flags {}", program, extra_flags.concat());
     }
 
+    #[rstest]
+    #[case(
+        "dict_with_struct_non_squash.cairo",
+        "{0: 1 true 1: 1 false 2: 1 true}",
+        None
+    )]
+    #[case("nullable_box_vec_non_squash.cairo", "{0: 10 1: 20 2: 30} 3", None)]
+    #[case(
+        "felt_dict_non_squash.cairo",
+        "{66675: [8 9 10 11] 66676: [1 2 3]}",
+        None
+    )]
+    fn test_run_progarm_non_proof(
```

Comment:
Add a comment here explaining why we test these programs in a different way than the others